semgrep-core/src/engine/Eval_generic.ml

```diff
@@ -219,6 +219,17 @@ let rec eval env code =
           logger#info "regexp %s on %s return %s" re s (show_value v);
           v
       | _ -> raise (NotHandled code))
+  (* TERRAFORM:
+     The propagated value is stored in the `id_svalue` of the DotAccess, for Terraform.
+     We need to add this case here, or else we can't do metavariable analysis on an
+     identifier which contains a propagated value, because identifiers look like
+     DotAccesses in Terraform.
+  *)
+  | G.DotAccess
+      ( { e = G.N (Id ((("local" | "var"), _), _)); _ },
+        _,
+        FN (Id (_, { id_svalue = { contents = Some (Lit lit); _ }; _ })) ) ->
```

Comment:
There are already two cases that handle const-prop values, the normal case and the Dockerfile case, and I think you could add this case there too. Also this lacks `when env.constant_propagation ->`, although if you moved with the other two cases that would solve it already.